Fix DokuWiki 2025-05-14a compatibility issues

tryweb opened the pull request:
- Remove deprecated require_once() and fix undefined array keys
- Update DokuHTTPClient to namespaced class
- Fix undefined variables and unsafe array operations
- Improve GeSHi loading with fallback support

tryweb committed:
Fix DokuWiki 2025-05-14a compatibility issues

- Remove deprecated require_once() and fix undefined array keys
- Update DokuHTTPClient to namespaced class
- Fix undefined variables and unsafe array operations
- Improve GeSHi loading with fallback support

splitbrain commented on syntax.php:
```diff
@@ -21,7 +21,7 @@
 
 if(!defined('DOKU_PLUGIN')) define('DOKU_PLUGIN',DOKU_INC.'lib/plugins/');
-require_once(DOKU_PLUGIN.'syntax.php');
+// require_once(DOKU_PLUGIN.'syntax.php'); // Removed as per Dokuwiki 2025 autoloading
```
please don't comment out code. remove it

tryweb replied:
Removed the commented code as requested.

splitbrain commented on syntax.php:
```diff
@@ -64,7 +69,7 @@ function render($mode, Doku_Renderer $renderer, $data) {
         $base  = hsc($data[0]);
         $title = ($data[1] ? hsc($data[1]) : $base);
-        $path  = hsc($_REQUEST['repo']);
+        $path  = hsc($_REQUEST['repo'] ?? '');
```
global $INPUT should be used instead of accessing $_REQUEST directly

tryweb replied:
Fixed! Replaced $_REQUEST with global $INPUT as per DokuWiki coding standards.

splitbrain commented on syntax.php:
```diff
@@ -41,8 +41,13 @@ function connectTo($mode) {
         $match = substr($match, 7, -2);
-        list($base, $title) = explode('|', $match, 2);
-        list($base, $refresh) = explode(' ', $base, 2);
+        $parts = explode('|', $match, 2);
```
use `sexplode()` function instead.

tryweb replied:
Replaced all explode() calls with sexplode() in commit 94377a5c1a4dc2bdca0c9878978aac3adf9417bb

splitbrain commented on syntax.php:
```diff
@@ -122,7 +127,7 @@ function _directory($url, &$renderer, $path, $refresh) {
 
         // download the index html file
-        $http = new DokuHTTPClient();
+        $http = new \dokuwiki\HTTP\DokuHTTPClient();
```
Use a `use` import instead.

tryweb replied:
Added use import statement

splitbrain commented on syntax.php:
```diff
@@ -182,31 +187,37 @@ function _cached_geshi($url, $refresh) {
+                } elseif (file_exists(DOKU_INC .'inc/geshi.php')) {
+                    require_once(DOKU_INC .'inc/geshi.php');
+                }
+            }
```
Can completely be removed. Autoloading will take car of loading GeSHi

tryweb replied:
Done! Removed the manual GeSHi include since autoloading handles it.

splitbrain reviewed (changes requested):
Could you please look at my comments and improve the code accordingly?

Klap-in commented on syntax.php:
```diff
@@ -21,7 +21,7 @@
 if(!defined('DOKU_INC')) die();
 
 if(!defined('DOKU_PLUGIN')) define('DOKU_PLUGIN',DOKU_INC.'lib/plugins/');
```
Please remove the die() and define DOKU_PLUGIN as well, it is old code.

tryweb replied:
Removed die() and DOKU_PLUGIN define - legacy code cleanup

tryweb committed:
refactor(syntax): apply code review improvements

- feat: use $INPUT instead of $_REQUEST for parameter handling
- refactor: replace explode() with sexplode() for consistency
- style: add use statement for DokuHTTPClient
- cleanup: remove manual GeSHi autoloading code
- cleanup: remove deprecated die() and DOKU_PLUGIN definitions

Addresses multiple code review suggestions to modernize the codebase
and follow DokuWiki best practices.

splitbrain commented on syntax.php:
```diff
@@ -41,8 +37,13 @@ function connectTo($mode) {
+        
+        $parts = sexplode(' ', $base, 2);
+        $base = $parts[0] ?? '';
+        $refresh = $parts[1] ?? '';
```
the point of sexplode is that it will ensure to return the right amount of elements. So this can simply be `[$base, $refresh] = sexplode(' ', $base, 2);`

tryweb replied:
Great point! Updated to use array destructuring - much cleaner. Thanks!

splitbrain commented on syntax.php:
```diff
@@ -60,11 +61,12 @@ function handle($match, $state, $pos, Doku_Handler $handler) {
      * Create output
      */
     function render($mode, Doku_Renderer $renderer, $data) {
```
sorry forgot to mention. please declare proper visibility (eg. public here)

tryweb replied:
Done! Added proper visibility declaration to the render method.

splitbrain commented on syntax.php:
```diff
@@ -90,19 +92,19 @@ function render($mode, Doku_Renderer $renderer, $data) {
     /**
      * Handle remote directories
      */
     function _directory($url, &$renderer, $path, $refresh) {
```
should be `protected directory()` (PSR-2)

tryweb replied:
Done! Updated to protected visibility and added underscore prefix per PSR-2.

splitbrain commented on syntax.php:
```diff
@@ -228,7 +228,7 @@ function _location($path, $title, &$renderer) {
 
         $base = '';
-        $dirs = explode('/', $path);
+        $dirs = sexplode('/', $path);
```
no need for sexplode when you dont't specify the number of expected results.

tryweb replied:
Reverted to explode() since no limit specified

splitbrain reviewed (changes requested):
nearly there. just a few minor things. Thanks for your help!

tryweb committed:
refactor: use array destructuring for sexplode() calls

tryweb committed:
refactor: add proper visibility modifiers to all methods

tryweb committed:
refactor: change private methods to protected per PSR-2

tryweb committed:
refactor: use explode when not limiting result count

splitbrain reviewed (commented):
Methods are still not PSR-2 compatible. Eg `protected function _cached_geshi()`should become `protected function cachedGeshi()`.

I was about to apply this simple fix myself, but am having a hard time to get the plugin working at all. Can you please update the plugin page? Remove all the old cruft and add example how to use the plugin with common code foundries like github.
